src/search/cost_test_ebpf.cc

```diff
@@ -214,8 +299,211 @@ void test1() {
+                inst(),
+                inst(),
+               };
+
+
+  inst p2[] = { inst(MOV64XC, 0, 1), //Moving 0 into register r0
+                inst(STXB, 1, 0, 0), // *(u8 *)(pkt + 0) = r0
+                inst(STXB, 1, 1, 0), // *(u8 *)(pkt + 1) = r0
+                inst(STXB, 1, 2, 0), // *(u8 *)(pkt + 2) = r0
+                inst(STXB, 1, 3, 0), // *(u8 *)(pkt + 3) = r0
+                //inst(MOV64XC, 0, 0),
+                inst(),
+                inst(),
+               };
+  
+
+
+  int win_start = 0, win_end = 0;
+
+  mem_t::_layout.clear();
+  //use sizeof(program)/sizeof(ins) for below
+  inst::max_prog_len = sizeof(p1) / sizeof(inst);
+
+  //inst_var.h (line 46): PGM_INPUT types for different kinds of program inputs
+  mem_t::set_pgm_input_type(PGM_INPUT_pkt);
+  //we don't use packet in the above programs
+  mem_t::set_pkt_sz(5);
+  //mem_t::add_map(map_attr(128, 64, 91));
+  //mem_t::add_map(map_attr(96, 96, 91));
+  //mem_t::add_map(map_attr(64, 128, 91));
+  mem_t::_layout._n_randoms_u32 = 1;
+  smt_var::init_static_variables();
+
+  print_test_res(get_error_cost_repair(p1, p2, win_start, win_end) > 0, "error_cost_repair: 2 safe and non-equivalent programs ");
+  print_test_res(get_safety_cost_repair(p1, p2, win_start, win_end) == 0, "safety_cost_repair: 2 safe and non-equivalent programs ");
+  mem_t::_layout.clear();
+
+}
+
+void test6(){
+
+  //orig: safe program
+  //synth: unsafe program
+  //safety_cost_repair should be greater than 0.
+
+  // *(u32 *)pkt = 0
+  //r1 is pkt start address
+  inst p1[] = { inst(MOV64XC, 0, 0),
+                inst(STXB, 1, 0, 0), // *(u8 *)(pkt + 0) = r0
+                inst(STXB, 1, 1, 0), // *(u8 *)(pkt + 1) = r0
+                inst(STXB, 1, 2, 0), // *(u8 *)(pkt + 2) = r0
+                inst(STXB, 1, 3, 0), // *(u8 *)(pkt + 3) = r0
+                inst(),
+                inst(),
+               };
+
+  inst p2[] = {inst(STXB, 1, 0, 0), // *(u8 *)(pkt + 0) = r0
+                inst(STXB, 1, 1, 0), // *(u8 *)(pkt + 1) = r0
+                inst(STXB, 1, 2, 0), // *(u8 *)(pkt + 2) = r0
+                inst(STXB, 1, 3, 0), // *(u8 *)(pkt + 3) = r0
+                inst(MOV64XC, 0, 0),
+                inst(),
+                inst(),
+              };
+
+  int win_start = 0, win_end = 4;
+
+  mem_t::_layout.clear();
+  //use sizeof(program)/sizeof(ins) for below
+  inst::max_prog_len = sizeof(p1) / sizeof(inst);
+
+  //inst_var.h (line 46): PGM_INPUT types for different kinds of program inputs
+  mem_t::set_pgm_input_type(PGM_INPUT_pkt);
+  //we don't use packet in the above programs
+  mem_t::set_pkt_sz(5);
+  //mem_t::add_map(map_attr(128, 64, 91));
+  //mem_t::add_map(map_attr(96, 96, 91));
+  //mem_t::add_map(map_attr(64, 128, 91));
+  mem_t::_layout._n_randoms_u32 = 1;
+  smt_var::init_static_variables();
+
+  print_test_res(get_safety_cost_repair(p1, p2, win_start, win_end) > 0, "safety_cost_repair: 2 safe and non-equivalent programs ");
```

Comment:
Seems p1 and p2 in this test are safe and equivalent. Why get_safety_cost_repair is greater than 0? Is it supposed to be 0?